pkg/diff/diff.go

```diff
@@ -1087,11 +1071,20 @@ func toString(val interface{}) string {
 // Remarshalling also strips any type information (e.g. float64 vs. int) from the unstructured
 // object. This is important for diffing since it will cause godiff to report a false difference.
 func remarshal(obj *unstructured.Unstructured, o options) *unstructured.Unstructured {
-	obj = stripTypeInformation(obj)
 	data, err := json.Marshal(obj)
 	if err != nil {
 		panic(err)
 	}
+
+	// Unmarshal again to strip type information (e.g. float64 vs. int) from the unstructured
+	// object. This is important for diffing since it will cause godiff to report a false difference.
+	var newUn unstructured.Unstructured
+	err = json.Unmarshal(data, &newUn)
```

Comment:
This doesn't seem to provide the same results if compared with the original code. Previously we were stripping the types from `obj` and using that to marshal into `data`. In the new code, `data` will just hold the `Marshal(obj)` result which could lead to a different behaviour. To validate this, it would be better to write unit tests that validate the type stripping logic. Checking on Pod resources values will probably be an easy way to validate this:

```yaml
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "500m"
```

Reply:
`data` is only used only for another `Unmarshal` (or, more specifically, unmarshaling via `json.NewDecoder`, but the behavior should be the same).

So before we had:

```
unstructured -> marshal -> unmarshal to unstructured -> marshal -> unmarshal to schema type
```

Now it will be:

```
unstructured -> marshal -> unmarshal to schema type
```

i.e. we're skipping `unmarshal to unstructured` (which is just `map[string]interface{}`) and the second `marshal`.

My belief is that those to steps don't remove any type information that `unstructured -> marshal` already removes.

But I'll write some tests to give us a bit more confidence that we're not losing anything important.